src/magic_enchantment.cpp

```diff
@@ -631,10 +720,25 @@ void enchantment::cast_enchantment_spell( Character &caster, const Creature *tar
     }
 }
 
-bool enchantment::operator==( const enchantment &rhs ) const
+bool enchant_cache::operator==( const enchant_cache &rhs ) const
 {
+    if( this->values_add.size() != rhs.values_add.size() ) {
+        return false;
+    }
+    auto iter_add = this->values_add.cbegin();
+    auto iter_add2 = rhs.values_add.cbegin();
+    while( iter_add != this->values_add.cend() && iter_add2 != rhs.values_add.cend() ) {
```

Comment:
The loop does not change iter_add nor iter_add2 nor anything at all. The loop is either skipped (because the condition is false) or infinite (because the condition is true and never changes).

Same for below.

And you don't check whether values_multiply and rhs.values_multiply have the same size, but that is probably required.